pennylane/transforms/dynamic_one_shot.py

```diff
@@ -194,18 +205,26 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
                 new_measurements.append(SampleMP(obs=m.obs))
             else:
                 new_measurements.append(m)
-    for op in circuit:
-        if is_mcm(op):
+    new_operations = []
+    for op in circuit.operations:
+        if isinstance(op, MidMeasureMP):
             new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
-
+        elif "MidCircuitMeasure" in str(type(op)):
```

Comment:
This needs to be `if` because the block changes both `new_measurements` and `new_operations`.